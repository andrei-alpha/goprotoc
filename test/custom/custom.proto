package test;

import "github.com/dropbox/goprotoc/gogoproto/gogo.proto";

option (gogoproto.populate_all) = true;
option (gogoproto.equal_all) = true;
option (gogoproto.verbose_equal_all) = true;
option (gogoproto.stringer_all) = true;

message MessageCustom {
	optional int64 field1 = 1 [(gogoproto.customtype) = "github.com/dropbox/goprotoc/test.Id"];
	optional double field2 = 2 [(gogoproto.customtype) = "github.com/dropbox/goprotoc/test.Weight"];
	optional bool field3 = 3 [(gogoproto.customtype) = "github.com/dropbox/goprotoc/test.Truth"];
	optional string field4 = 4 [(gogoproto.customtype) = "github.com/dropbox/goprotoc/test.Sbring"];
	optional bytes field5 = 5 [(gogoproto.customtype) = "github.com/dropbox/goprotoc/test.Obj"];

	repeated int64 field11 = 11 [(gogoproto.customtype) = "github.com/dropbox/goprotoc/test.Id"];
	repeated double field12 = 12 [(gogoproto.customtype) = "github.com/dropbox/goprotoc/test.Weight"];
	repeated bool field13 = 13 [(gogoproto.customtype) = "github.com/dropbox/goprotoc/test.Truth"];
	repeated string field14 = 14 [(gogoproto.customtype) = "github.com/dropbox/goprotoc/test.Sbring"];
	repeated bytes field15 = 15 [(gogoproto.customtype) = "github.com/dropbox/goprotoc/test.Obj"];
}
