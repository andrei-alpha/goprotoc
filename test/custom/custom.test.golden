package test

import (
	"testing"
	"time"
	math_rand "math/rand"

	"github.com/dropbox/goprotoc/proto"
)

func TestCustom(t *testing.T) {
	popr := math_rand.New(math_rand.NewSource(time.Now().UnixNano()))
	p := NewPopulatedMessageCustom(popr, false)
	data, err := proto.Marshal(p)
	if err != nil {
		panic(err)
	}
	msg := &MessageCustom{}
	if err := proto.Unmarshal(data, msg); err != nil {
		panic(err)
	}
	for i := range data {
		data[i] = byte(popr.Intn(256))
	}
	if err := p.VerboseEqual(msg); err != nil {
		t.Fatalf("%#v !VerboseProto %#v, since %v", msg, p, err)
	}
	if !p.Equal(msg) {
		t.Fatalf("%#v !Proto %#v", msg, p)
	}
}
